### ISSUE TITLE
Allow `sc.profile()` to trace calls recursively

### ISSUE THREAD

**Issue by cliffckerr** [sciris/sciris#604]
Allow a module to be provided to `follow` instead of a (list of) functions, and add a `recursive=True` argument.

Examples:
```py
import starsim as ss

sim = ss.Sim()

sc.profile(run=sim.run, follow=ss.sim) # Follows all functions in ss.sim (i.e. starsim/sim.py)
sc.profile(run=sim.run, follow=ss, top=10, mintime=1e-2) # Follows the top 10 longest-running functions in the Starsim module
sc.profile(run=sim.run, follow=sim.run, recursive=True) # Follow sim.run and all subfunctions
```

**Comment by cliffckerr** [comment]
Also allow following by (sub)class, e.g.
```py
sc.profile(run=sim.run, follow=ss.Disease, recursive=False # Follow only functions in ss.Disease
sc.profile(run=sim.run, follow=ss.Disease, recursive=True) # Follow functions in ss.Disease and any subclasses
```

**Comment by cliffckerr** [comment]
Closed by #613
